packages/nextclade-web/src/io/fetchSingleDatasetAuspice.ts

```diff
@@ -1,13 +1,24 @@
+import { isEmpty } from 'lodash'
 import { attrStrMaybe, AuspiceTree, Dataset } from 'src/types'
 import { removeTrailingSlash } from 'src/io/url'
-import { axiosFetch } from 'src/io/axiosFetch'
+import { axiosFetch, axiosFetchOrUndefined } from 'src/io/axiosFetch'
 
 export async function fetchSingleDatasetAuspice(datasetJsonUrl_: string) {
   const datasetJsonUrl = removeTrailingSlash(datasetJsonUrl_)
 
   const auspiceJson = await axiosFetch<AuspiceTree>(datasetJsonUrl, {
     headers: { Accept: 'application/json, text/plain, */*' },
   })
+
+  if (isEmpty(auspiceJson.root_sequence)) {
+    const sidecar = await axiosFetchOrUndefined<Record<string, string>>(datasetJsonUrl, {
+      headers: { Accept: 'application/vnd.nextstrain.dataset.root-sequence+json' },
+    })
+    if (!isEmpty(sidecar)) {
+      auspiceJson.root_sequence = sidecar
+    }
+  }
```

Comment:
Is this correct to just take the sidecar contents and merge it into `.root_sequence`, like if it always was there?

Reply:
Yup! ([Here's the relevant augur code](https://github.com/nextstrain/augur/blob/f0900bce1a51c4c82b81486b2bcc709db6c08239/augur/export_v2.py#L1239-L1248))

Reply:
And the [v2 dataset schema](https://nextstrain.org/schemas/dataset/v2) [references the root-sequence sidecar schema](https://github.com/nextstrain/augur/blob/f0900bce1a51c4c82b81486b2bcc709db6c08239/augur/data/schema-export-v2.json#L124-L126) for the `.root_sequence` field.